### PR TITLE
AND/OR Short circuit

### PR DESCRIPTION
**Pull request by chhatch**
Closes - [OR/AND operator must short-circuit on falsey condition](https://github.com/elite-libs/rules-machine/issues/38)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -166,11 +171,16 @@ export function ruleFactory<
...
+          const or = arrayify(rule.if.or);
+          for (const rule of or) {
+            // @ts-expect-error
+            conditionResult = evaluateRule({
+              stepRow,
+              input,
+              rule,
+            });
+            if (conditionResult) break;
+          }
```
Comment (chhatch) [inline src/index.ts:183]: @justsml is this sort of what you had in mind?

Comment (chhatch) [inline src/index.ts]: @justsml is this what you had in mind?

Comment (justsml) [reply src/index.ts]: I think that does it 🤘 Nice work!

File: src/pr.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      { if: { and: ['foo', 'bar = 42'] }, then: '1' },
+      { return: 'bar' },
+    ];
+
+    expect(ruleFactory(rules)({ foo: false })).toBe(undefined);
+    expect(ruleFactory(rules)({ foo: true })).toBe(42);
+  });
+
+  it('"and" string rule should short circuit', () => {
+    const rules = [{ if: 'foo and bar = 42', then: '1' }, { return: 'bar' }];
```
Comment (justsml) [inline src/pr.test.ts]: This might need to be a different PR, as the string expression parsing stuff is handled differently than the structural object stuff.

**Comment by chhatch** [comment]
Still need to update docs to note the difference in behavior of `AND` vs `{and: ...}`
